tests/components/madvr/test_sensors.py

```diff
@@ -0,0 +1,264 @@
+"""Tests for the MadVR sensor entities."""
+
+from __future__ import annotations
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.const import Platform, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.entity_registry as er
+
+from . import setup_integration
+from .conftest import get_update_callback
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+async def test_sensor_setup(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test setup of the sensor entities."""
+    with patch("homeassistant.components.madvr.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+@pytest.mark.parametrize(
```

Comment:
In this case, might as well just create one big data update and call this before the snapshot and make the snapshot snapshot all the updated states

Reply:
In the previous test